### PR TITLE
feat: revamp python docker templates

### PR DESCRIPTION
**Pull request by tanmoysrt**
Related #55 #56

#### Describe the changes you have made in this PR -
- Make it generic, just by changing `SETUP_COMMAND` and `START_COMMAND` make it compatible for different python frameworks
- OS package installation support added
- Added support for running separate commands if the user wants

#### Tested on
- [x] Flask Pip -> https://github.com/tanmoysrt/geenie
- [x] Flask Pipenv -> https://github.com/tanmoysrt/geenie
- [x] Flask Poetry -> https://github.com/NarayanAdithya/Flask-Poetry
- [x] Django Pip -> https://github.com/tanmoysrt/django_blog/tree/master/blog
- [x] Django Pipenv -> https://github.com/tanmoysrt/django_blog/tree/master/blog
- [x] Django Poetry -> https://github.com/tanmoysrt/django_blog/tree/master/blog
- [x] Streamlit Pip -> https://github.com/patrickloeber/streamlit-demo
- [x] Streamlit Pipenv -> https://github.com/patrickloeber/streamlit-demo
- [x] Streamlit Poetry -> https://github.com/patrickloeber/streamlit-demo
- [x] Fastapi Pip -> https://github.com/sumanentc/python-sample-FastAPI-application
- [x] Fastapi Pipenv -> https://github.com/sumanentc/python-sample-FastAPI-application
- [x] Fastapi Poetry -> https://github.com/sumanentc/python-sample-FastAPI-application

Note: Please check **Allow edits from maintainers.** before creating the PR.

### CODE REVIEW

File: docker_config_generator/templates/python-poetry.Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
-FROM python:3.10 as python-base
+FROM python:3.10-bullseye as python-base
```
Comment (The-Debarghya) [inline docker_config_generator/templates/python-poetry.Dockerfile:1]: `python:3.11-bullseye` would be conflict-less Ig, w.r.t PR #58

Comment (tanmoysrt) [reply docker_config_generator/templates/python-poetry.Dockerfile:1]: Ok 👌we can update that

**Review by The-Debarghya — APPROVED** [review]

**Comment by tanmoysrt** [comment]
@all-contributors 
please add @tanmoysrt for infra 
please add @tanmoysrt for review
please add @The-Debarghya  for review
please add @cAmartya   for review

**Comment by tanmoysrt** [comment]
@all-contributors
please add @cAmartya for review

**Pull request merged**